extension/src/fileSystem/watcher.ts

```diff
@@ -34,18 +34,18 @@ export const fireWatcher = (path: string): Promise<void> => {
 export const ignoredDotDirectories = /.*[/\\|]\.(dvc|(v)?env)[/\\|].*/
 
 export const createFileSystemWatcher = (
+  track: (disposable: Disposable) => Disposable,
   glob: GlobPattern,
   listener: (path: string) => void
-): Disposable => {
+): void => {
   if (typeof glob === 'string' && isDirectory(glob)) {
     throw new Error(
       'FileSystemWatcher will not behave as expected under these circumstances.'
     )
   }
   const fileSystemWatcher = workspace.createFileSystemWatcher(glob)
-  fileSystemWatcher.onDidCreate(uri => listener(uri.fsPath))
-  fileSystemWatcher.onDidChange(uri => listener(uri.fsPath))
-  fileSystemWatcher.onDidDelete(uri => listener(uri.fsPath))
-
-  return fileSystemWatcher
+  track(fileSystemWatcher)
+  track(fileSystemWatcher.onDidCreate(uri => listener(uri.fsPath)))
+  track(fileSystemWatcher.onDidChange(uri => listener(uri.fsPath)))
+  track(fileSystemWatcher.onDidDelete(uri => listener(uri.fsPath)))
```

Comment:
[F] The difference here is that previously I expected that disposing the `fileSystemWatcher` meant that all of the listeners were disposed as well. I now think that this might not actually be the case. Just to be sure I have started tracking the listeners.